web/src/components/organisms/Project/Content/ViewsMenu/hooks.ts

```diff
@@ -16,42 +18,63 @@ import { CurrentViewType } from "../ContentList/hooks";
 type Params = {
   modelId?: string;
   currentView: CurrentViewType;
-  // setCurrentView: (view: CurrentViewType) => void;
+  setCurrentView: (currentView: CurrentViewType) => void;
 };
 
-export default ({ modelId, currentView }: Params) => {
+export type modalStateType = "rename" | "create";
+
+export default ({ modelId, currentView, setCurrentView }: Params) => {
   const t = useT();
   const [viewModalShown, setViewModalShown] = useState(false);
-  const [selectedView, setSelectedView] = useState<View>();
+  const [selectedView, setSelectedView] = useState<View | undefined>();
+  const [modalState, setModalState] = useState<modalStateType>("create");
   const [submitting, setSubmitting] = useState(false);
   const [currentProject] = useProject();
 
   const projectId = useMemo(() => currentProject?.id, [currentProject]);
 
-  const { data } = useGetViewsQuery({
+  const { data, loading } = useGetViewsQuery({
     variables: { modelId: modelId ?? "" },
     skip: !modelId,
   });
 
   const views = useMemo(() => {
+    if (loading && data?.view) {
+      setSelectedView(data?.view && data?.view.length > 0 ? (data?.view[0] as View) : undefined);
+    }
     return data?.view ? data?.view : [];
-  }, [data?.view]);
+  }, [data?.view, loading]);
+
+  useEffect(() => {
+    setSelectedView(data?.view && data?.view.length > 0 ? (data?.view[0] as View) : undefined);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [modelId, setCurrentView]);
+
+  useEffect(() => {
+    if (selectedView) {
+      setCurrentView({
+        sort: selectedView.sort as ItemSort,
+        columns: selectedView.columns as FieldSelector[],
+      });
+    } else {
+      setCurrentView({ columns: [] });
+    }
+  }, [selectedView, setCurrentView]);
```

Comment:
As much as possible try to avoid Type assertions as they lead into a lot of bugs in the future.
`xxx as someType`